contributors/design-proposals/flexvolume-deployment.md

```diff
@@ -0,0 +1,148 @@
+# **Automated Flexvolume Deployment**
+
+## **Objective**
+
+Automate the deployment of Flexvolume drivers with the following goals:
+*   Drivers must be deployed on nodes (and master when attach is required) without having to manually access any machine instance.
+*   Kubelet and controller-manager do not need to be restarted manually in order for the new plugin to be recognized.
+
+## **Background**
+
+Beginning in version 1.8, the Kubernetes Storage SIG is putting a stop to accepting in-tree volume plugins and advises all storage providers to implement out-of-tree plugins. Currently, there are two recommended implementations: Container Storage Interface (CSI) and Flexvolume.
+
+[CSI](https://github.com/container-storage-interface/spec/blob/master/spec.md) provides a single interface that storage vendors can implement in order for their storage solutions to work across many different container orchestrators, and volume plugins are out-of-tree by design. This is a large effort, the full implementation of CSI is several quarters away, and there is a need for an immediate solution for storage vendors to continue adding volume plugins.
+
+[Flexvolume](https://github.com/kubernetes/community/blob/master/contributors/devel/flexvolume.md) is an in-tree plugin that has the ability to run any storage solution by executing volume commands against a user-provided driver on the Kubernetes host, and this currently exists today. However, the process of setting up Flexvolume is very manual, pushing it out of consideration for many users. Problems include having to copy the driver to a specific location in each node, manually restarting kubelet, and user's limited access to machines.
+
+
+## **Overview**
+
+
+### User Story
+
+Driver Installation:
+
+*   Alice is a storage plugin author and would like to deploy a Flexvolume driver on all node instances. She
+    1. prepares her Flexvolume driver directory, with driver names in `[vendor~]driver/driver` format (e.g. `flexvolume/k8s~nfs/nfs`, see [Flexvolume documentation](https://github.com/kubernetes/community/blob/master/contributors/devel/flexvolume.md#prerequisites)).
+    2. creates an image by copying her driver to the Flexvolume deployment base image at `/flexvolume`.
```

Comment:
it might simplify this proposal to focus it on dynamic discovery of new plugins, and update of existing plugins. deployment via daemonset could be one mechanism, but shouldn't be prescriptive.